File: app/src/main/java/com/vinicius/receitasapp/RecipeAdapter.java
package com.vinicius.receitasapp;

import android.content.Context;
import android.content.Intent;
import android.view.LayoutInflater;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.vinicius.receitasapp.databinding.RecipeItemBinding;

import java.util.List;

public class RecipeAdapter extends RecyclerView.Adapter<RecipeAdapter.ViewHolder> {

    private final List<Receitas> recipes;

    public RecipeAdapter(List<Receitas> receitas) {
        this.recipes = receitas;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        LayoutInflater layoutInflater = LayoutInflater.from(parent.getContext());
        RecipeItemBinding binding = RecipeItemBinding.inflate(layoutInflater, parent, false);
        return new ViewHolder(binding);
    }

    @Override
    public void onBindViewHolder(@NonNull RecipeAdapter.ViewHolder holder, int position) {
        Receitas recipe = recipes.get(position);
        Context context = holder.itemView.getContext();

        holder.binding.recipeName.setText(recipe.getNome());
        holder.binding.amountPeopleRecipe.setText(recipe.getPorcoes());
        holder.binding.difficultyRecipe.setText(recipe.getDificuldade());
        holder.binding.timeRecipe.setText(recipe.getTempo());

        holder.itemView.setOnClickListener(view -> {
            Intent intent = new Intent(context, RecipeActivity.class);
            intent.putExtra(RecipeActivity.Extras.RECIPE, recipe);
            context.startActivity(intent);
        });

    }

    @Override
    public int getItemCount() {
        return recipes.size();
    }

    public static class ViewHolder extends RecyclerView.ViewHolder {

        private final RecipeItemBinding binding;

        public ViewHolder(RecipeItemBinding binding) {
            super(binding.getRoot());
            this.binding = binding;
        }
    }
}

File: app/src/main/java/com/vinicius/receitasapp/ui/RecipeAdapter.kt
package com.vinicius.receitasapp.ui

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.DiffUtil
import androidx.recyclerview.widget.ListAdapter
import androidx.recyclerview.widget.RecyclerView
import com.vinicius.receitasapp.data.Recipes
import com.vinicius.receitasapp.databinding.RecipeItemBinding


class RecipeAdapter : ListAdapter<Recipes, RecipeAdapter.ViewHolder>(DiffCallBack()) {

        var listenerShare: (View) ->Unit ={}

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val inflater = LayoutInflater.from(parent.context)
        val binding = RecipeItemBinding.inflate(inflater, parent, false)
        return ViewHolder(binding)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.bind(getItem(position))
    }

    inner class ViewHolder(private val binding: RecipeItemBinding) : RecyclerView.ViewHolder(binding.root) {
        fun bind(recipe:Recipes){
            binding.tvRecipeName.text = recipe.name
            binding.tvRecipeAmount.text = recipe.amount
            binding.tvRecipeTime.text = recipe.time
            binding.tvRecipeAmount.text = recipe.amount
            binding.cdContent.setOnClickListener {
                listenerShare(it)
            }
            //binding.ivRecipeImage.setImageResource(R.drawable.rice_img)

        }

    }


}

class DiffCallBack: DiffUtil.ItemCallback<Recipes>() {
    override fun areItemsTheSame(oldItem: Recipes, newItem: Recipes) = oldItem == newItem

    override fun areContentsTheSame(oldItem: Recipes, newItem: Recipes) = oldItem.id == newItem.id

}
